happytransformer/happy_generation.py

```diff
@@ -0,0 +1,163 @@
+"""
+Contains the HappyGeneration class
+"""
+from dataclasses import dataclass
+from transformers import AutoModelForCausalLM
+from happytransformer.happy_transformer import HappyTransformer
+from happytransformer.toc.trainer import TOCTrainer
+from happytransformer.adaptors import get_adaptor
+
+DEFAULT_SETTINGS = {
+    "do_sample": False,
+    "early_stopping": False,
+    "num_beams": 1,
+    "temperature": 0.65,
+    "top_k": 50,
+    "top_p": 1.0,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
+}
+
+default_greedy_settings = {
+    "do_sample": False,
+    "early_stopping": False,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
+}
+
+
+default_beam_settings = {
+    "num_beams": 5,
+    "early_stopping": True,
+
+    "do_sample": False,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
+}
+
+default_generic_sampling_settings = {
+    "do_sample": True,
+    "temperature": 0.7,
+    "top_k": 0,
+    "early_stopping": False,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
+}
+
+
+default_top_k_sampling_settings = {
+    "do_sample": True,
+    "top_k": 50,
+    "early_stopping": False,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
+}
+
+
+default_p_nucleus_sampling_settings = {
+    "do_sample": True,
+    "top_k": 0,
+    "top_p": 0.92,
+    "early_stopping": False,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
+}
+
+
+@dataclass
+class GenerationResult:
+    text: str
+
+
+class HappyGeneration(HappyTransformer):
+    """
+    A user facing class for text generation
+    """
+    def __init__(self, model_type: str = "GPT2", model_name: str = "gpt2"):
+
+        self.adaptor = get_adaptor(model_type)
+
+        model = AutoModelForCausalLM.from_pretrained(model_name)
+
+        super().__init__(model_type, model_name, model)
+
+        self._trainer = TOCTrainer(self.model, model_type, self.tokenizer, self._device, self.logger)
+
+    def __check_default_text_is_val(self, text):
```

Comment:
"default" seems an odd word for a user supplied value. I would propose `_assert_starting_text_is_valid()`

Reply:
Fixed in a1ada70